packages/core/src/browser/shell/application-shell.ts

```diff
@@ -1744,8 +1744,6 @@ export class ApplicationShell extends Widget {
         if (!prevBar) {
             return false;
         }
-        const len = prevBar.titles.length;
-        prevBar.currentIndex = len - 1;
```

Comment:
I like the elegance of the solution :-). Perhaps, rather than removing this code completely, it should be put in a guard:

```typescript
if (!prevBar.currentTitle) {
    prevBar.currentIndex = prevBar.title.length - 1;
}
```
I'm not certain under what conditions `currentTitle` would be undefined, but this way we ensure that some widget on the previous bar is activated, but we don't change which widget is active if one is already marked so.

Reply:
sounds good, added the mentioned guard :+1: